### DISCUSSION TITLE
SharpGl not work

### DISCUSSION THREAD

**Discussion by tqk2811** [MasterAler/SampleYUVRenderer#6] [Show and tell]
Firstly thank you.  
That `frame` i got from libav with pixel format `YUV420P`  
`Console Output:  Pushed 3196433, WxH: 1080x2400 linesize:1152,576,576`  
 ```
public unsafe void PushFrame(AVFrame* frame)
{
  switch((AVPixelFormat)frame->format)
  {
    //case AVPixelFormat.AV_PIX_FMT_NV12://VGA decode
    //  break;
    case AVPixelFormat.AV_PIX_FMT_YUV420P://h264 decode
      break;
    default: throw new NotSupportedException(((AVPixelFormat)frame->format).ToString());
  }

  AVFrame* clone = av_frame_clone(frame);
  frames.Enqueue(clone);
  Console.WriteLine($"Pushed {clone->pts}, WxH: {clone->width}x{clone->height} linesize:{clone->linesize[0]},{clone->linesize[1]},{clone->linesize[2]}");
}
```

Can you look a bit my code? Thank.  
[InitShader](https://github.com/tqk2811/TqkLibrary.Media.VideoPlayer/blob/61b83da697d7ca48ac59227547848482fa63a757/TqkLibrary.Media.VideoPlayer.OpenGl/OpenGlVideoPlayer.xaml.cs#L125) (sub function and shader string: [BuildShader](https://github.com/tqk2811/TqkLibrary.Media.VideoPlayer/blob/61b83da697d7ca48ac59227547848482fa63a757/TqkLibrary.Media.VideoPlayer.OpenGl/ShadersHelper.cs#L40))  it like `OpenGLDisplay::initializeGL`  
[RenderFrame](https://github.com/tqk2811/TqkLibrary.Media.VideoPlayer/blob/61b83da697d7ca48ac59227547848482fa63a757/TqkLibrary.Media.VideoPlayer.OpenGl/OpenGlVideoPlayer.xaml.cs#L162) it like `OpenGLDisplay::paintGL`

And screen output only black

**Comment by MasterAler** [comment]
@tqk2811 

Hmmm... let's begin with simple assumptions:

1. area shaders the same?
2. no errors in the output? `glGetError` returns something if inserted for debug purposes?
3. if you take _exacly_ the same unmanaged code, even without for loop, as close as possible -- any changes then?

UPD: ah, haven't noticed the Closed status. Glad that it's been solved!
Feel free to share any tips or propose code improvements))

**Comment by tqk2811** [comment]
Thankyou again.
I did it, [YUV420Render](https://github.com/tqk2811/TqkLibrary.Media.VideoPlayer/blob/867335c207b7fb6c16dc57f077d0537fa328b386/TqkLibrary.Media.VideoPlayer.OpenGl/Renders/YUV420Render.cs#L14)  
![image](https://user-images.githubusercontent.com/13986296/131022838-f7258a92-8a2a-4f3b-ac12-99b923c9dbbf.png)  
________________________
seems like color a bit white
![image](https://user-images.githubusercontent.com/13986296/131023053-4e0dd94e-9882-4b38-b356-c704c4bc5ef0.png)
